app/src/main/java/com/framgia/englishconversation/data/source/remote/timeline/TimelineRemoteDataSource.java

```diff
@@ -22,62 +25,137 @@
+        });
+    }
+
+    @Override
+    public Observable<List<TimelineModel>> getTimeline(final TimelineModel lastTimeline) {
+        return Observable.create(new ObservableOnSubscribe<List<TimelineModel>>() {
+            @Override
+            public void subscribe(final ObservableEmitter<List<TimelineModel>> e) throws Exception {
+                final Query query;
+                if (lastTimeline == null) {
+                    query = mReference.orderByChild(Constant.DatabaseTree.CREATED_AT)
+                            .limitToFirst(NUM_OF_TIMELINE_PER_PAGE);
+                } else {
+                    query = mReference.orderByChild(Constant.DatabaseTree.CREATED_AT)
+                            .startAt(-lastTimeline.getCreatedAt(), lastTimeline.getId())
+                            .limitToFirst(NUM_OF_TIMELINE_PER_PAGE);
+                }
+
+                query.addListenerForSingleValueEvent(new ValueEventListener() {
                     @Override
-                    public void onComplete(@NonNull Task<Void> task) {
-                        if (task.isSuccessful()) {
-                            callback.onGetDataSuccess(null);
+                    public void onDataChange(DataSnapshot dataSnapshot) {
+                        List<TimelineModel> timelineModels = new ArrayList<>();
+                        for (DataSnapshot snapshot : dataSnapshot.getChildren()) {
+                            TimelineModel timelineModel = snapshot.getValue(TimelineModel.class);
+                            timelineModel.setCreatedAt(
+                                    Utils.generateOppositeNumber(timelineModel.getCreatedAt()));
+                            timelineModel.setId(snapshot.getKey());
+                            if (lastTimeline == null || !lastTimeline.getId()
+                                    .equals(snapshot.getKey())) {
+                                timelineModels.add(timelineModel);
+                            }
                         }
+                        e.onNext(timelineModels);
                     }
-                })
-                .addOnFailureListener(new OnFailureListener() {
+
                     @Override
-                    public void onFailure(@NonNull Exception e) {
-                        callback.onGetDataFailed(e.getMessage());
+                    public void onCancelled(DatabaseError databaseError) {
+                        e.onError(databaseError.toException());
                     }
                 });
+            }
+        });
     }
 
     @Override
-    public void getTimeline(final TimelineCallback callback, final TimelineModel lastTimeline) {
+    public Observable<TimelineModel> updateTimeline(final TimelineModel lastTimeline) {
+        return Observable.create(new ObservableOnSubscribe<TimelineModel>() {
+            @Override
+            public void subscribe(final ObservableEmitter<TimelineModel> e) throws Exception {
+                if (lastTimeline == null) {
```

Comment:
ERROR: [CORRECTNESS][NP_NULL_ON_SOME_PATH] Possible null pointer dereference of TimelineRemoteDataSource$3.val$lastTimeline in com.framgia.englishconversation.data.source.remote.timeline.TimelineRemoteDataSource$3.subscribe(ObservableEmitter)
http://findbugs.sourceforge.net/bugDescriptions.html#NP_NULL_ON_SOME_PATH